clang/test/CodeGenHLSL/debug/rwbuffer_debug_info.hlsl

```diff
@@ -0,0 +1,11 @@
+// RUN: %clang  --driver-mode=dxc -Zi -Fc - -T cs_6_6 -O0 %s | FileCheck %s
+
+// CHECK: #dbg_declare(ptr [[ThisReg:%this\..*]], [[ThisMd:![0-9]+]],
+// CHECK-DAG: [[ThisMd]] = !DILocalVariable(name: "this", arg: 1, scope: !{{[0-9]+}}, type: ![[type:[0-9]+]], flags: DIFlagArtificial | DIFlagObjectPointer)
```

Comment:
What is the actual type this is specifying? My read is it specifies `float4` which isn't really what we want. We probably want it to instead become an opaque struct pointer.